frontend/plan/src/components/selector/Selector.js

```diff
@@ -72,7 +72,7 @@ function Selector(props) {
         if (courses.length === 1) {
             getCourse(courses[0].id);
         }
-    }, [courses, getCourse, isSearchingCourseInfo]);
+    }, [isSearchingCourseInfo]); // eslint-disable-line
```

Comment:
I haven't had time to look at the entire thing, but it is generally bad practice/an anti-pattern to exclude variables in the useEffect from the dependency array. Might have to restructure code to avoid this

Reply:
^ yeah I agree

Reply:
<img width="935" alt="Screen Shot 2020-01-27 at 11 28 03 AM" src="https://user-images.githubusercontent.com/14356289/73193173-2a104d80-40f8-11ea-800d-311825812342.png">

I tried something like this ^^ should work

Reply:
![Screen Shot 2020-01-27 at 2 16 12 PM](https://user-images.githubusercontent.com/14356289/73205704-934f8b00-410f-11ea-9b13-e989be038984.png)
Something like this should also work, then we'd remove the useEffect. 

My qualms with removing course from the dependency array is that course essentially becomes a stale state in the useEffect, and is dependent on the invariant that we need that function to be called only if we change the isSearchingCourseInfo boolean, which could be subjected to change.

Reply:
We should probably also dispatch updateSearch first. You'd still have to dispatch updateSearch because that sets the searchInfoLoading flag to false